plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolConnectorTest.java

```diff
@@ -0,0 +1,679 @@
+    {
+        return "tmp_trino_" + System.nanoTime();
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup setup)
+    {
+        if (typeUnsupported(setup.getTrinoTypeName())) {
+            return Optional.of(setup.asUnsupported());
+        }
+        if (setup.getTrinoTypeName().equals("char(3)") && setup.getSampleValueLiteral().equals("'ab'")) {
+            // Exasol fills char(3) with spaces, causing test to fail
+            return Optional.of(new DataMappingTestSetup("char(3)", "'abc'", "'zzz'"));
+        }
+        return Optional.of(setup);
+    }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        if (typeUnsupported(setup.sourceColumnType()) || typeUnsupported(setup.newColumnType())) {
+            return Optional.of(setup.asUnsupported());
+        }
+        if (setup.newColumnType().equals("integer")) {
+            // Exasol reports integer columns as decimal
+            setup = setup.withNewColumnType("decimal(10,0)");
+        }
+        if (setup.newColumnType().equals("bigint")) {
+            // Exasol reports bigint columns as decimal
+            setup = setup.withNewColumnType("decimal(19,0)");
+        }
+        if (setup.newValueLiteral().equals("CAST(CAST(SMALLINT '32767' AS smallint) AS integer)")) {
+            // Exasol returns Java type BigDecimal instead of Integer
+            setup = setup.withNewValueLiteral("CAST(SMALLINT '32767' AS decimal)");
+        }
+        if (setup.newValueLiteral().equals("CAST(CAST(2147483647 AS integer) AS bigint)")) {
+            // Exasol returns Java type BigDecimal instead of Long
+            setup = setup.withNewValueLiteral("CAST(2147483647 AS decimal)");
+        }
+        if (setup.newValueLiteral().equals("CAST(CAST(BIGINT '-2147483648' AS bigint) AS integer)")) {
+            // Exasol returns Java type BigDecimal instead of BigInteger
+            setup = setup.withNewValueLiteral("CAST(BIGINT '-2147483648' AS decimal)");
+        }
+        if (setup.sourceColumnType().equals("timestamp(3)") && setup.newColumnType().equals("timestamp(6)")) {
+            // Exasol does not support TIMESTAMP with microsecond precision
+            return Optional.of(setup.withNewColumnType("date").withNewValueLiteral("date '2020-02-12'"));
+        }
+        if (setup.sourceColumnType().equals("char(20)") && setup.newColumnType().equals("varchar")) {
+            // Exasol requires length for VARCHAR type
+            return Optional.of(setup.withNewColumnType("varchar(25)"));
+        }
+        if (setup.newColumnType().equals("decimal(38,3)")) {
+            // Exasol supports max. precision 36
+            return Optional.of(setup.withNewColumnType("decimal(36,3)"));
+        }
+        return Optional.of(setup);
+    }
+
+    private boolean typeUnsupported(String type)
```

Comment:
Please forgive me I have an allergy regarding boolean methods with negated names.
Could we call the method `typeSupported()` and invert the logic? 

If you still want convenient if-cases (e.g. in `filterDataMappingSmokeTestData()` you can then create another method 
```
private boolean typeUnsupported(String type) {
    return ! typeSupported(type);
}
````

Reply:
👍 ✅